CMakeLists.txt

```diff
@@ -288,16 +288,17 @@ else()
     message(WARNING "Currently running on Windows with no rust-utils file. Module code should work as expected, but client code may fail unexpectedly.")
 endif()
 
-# TODO(RSDK-10637): Currently, rust_utils doesn't build correctly for windows in CI, so don't declare the library unless a user has provided their own build
-if (NOT WIN32)
+if (NOT WIN32) # build `SHARED` on unix-based systems
```

Comment:
Building `STATIC` here caused tests to fail. So I reverted to `SHARED` (which we were already doing). But, building `SHARED` causes the builds to fail entirely on windows! So, we build `STATIC` on windows, below.

Reply:
I don't immediately recall, but I'll switch back to `STATIC` now and see if/how it fails!

Reply:
```
CMake Error at /__w/viam-cpp-sdk/viam-cpp-sdk/test_package/build/gcc-12-x86_64-14-release/generators/cmakedeps_macros.cmake:66 (message):
  Library 'viam_rust_utils' not found in package.  If 'viam_rust_utils' is a
  system library, declare it with 'cpp_info.system_libs' property
```